moonstone/parsers/counts/taxonomy/metaphlan3.py

```diff
@@ -13,11 +13,52 @@ class Metaphlan3Parser(TaxonomyCountsBase, BaseParser):
     NCBI_tax_column = 'NCBI_tax_id'
 
     def __init__(self, *args, **kwargs):
+        self.analysis_type = kwargs['analysis_type']
+        kwargs.pop('analysis_type')
         super().__init__(*args, parsing_options={'skiprows': 1}, **kwargs)
 
+    def remove_duplicates(self, df):
+        df = df.set_index('clade_name')
+        if self.analysis_type == 'rel_ab':
+            total = 99.9999
+        else:
+            total = df[df.index.map(lambda x: len(x.split('|'))) == 1].sum()
+
+        # dataframe at rank level
+        new_df = df.index.map(lambda x: len(x.split('|')))
```

Comment:
here the naming might be wrong, it corresponds more to new_df_index right?

Reply:
It's because I needed to create the index first, but I wanted to limit my number of different variables

Reply:
Yes I totally got it but do not worry about the number of variables, it is better to have too many variables with consistent names that make the code more readable :)